addons/product/models/product_uom.py

```diff
@@ -0,0 +1,31 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductUom(models.Model):
+    _description = 'Link between the products and the UoM'
+
+    uom_id = fields.Many2one('uom.uom', 'Unit of Measure', required=True, ondelete='cascade')
+    product_id = fields.Many2one('product.product', 'Product', required=True, ondelete='cascade')
+    name = fields.Char('Barcode', index='btree_not_null', required=True)
+    company_id = fields.Many2one('res.company', 'Company', required=True, default=lambda self: self.env.company)
+
+    _barcode_uniq = models.Constraint('unique(name)', 'A barcode can only be assigned to one packaging.')
```

Comment:
In theory you can have them in different companies then? See  _check_barcode_uniqueness in product